<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>938</width>
    <height>570</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="focusPolicy">
      <enum>Qt::TabFocus</enum>
     </property>
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="documentMode">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Socket</string>
      </attribute>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="geometry">
        <rect>
         <x>470</x>
         <y>250</y>
         <width>421</width>
         <height>261</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="title">
        <string>Send：</string>
       </property>
       <widget class="QTextBrowser" name="textSend">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>381</width>
          <height>181</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="puB_sent">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>230</y>
          <width>93</width>
          <height>28</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Send</string>
        </property>
       </widget>
      </widget>
      <widget class="QLineEdit" name="PortEdit">
       <property name="geometry">
        <rect>
         <x>540</x>
         <y>50</y>
         <width>81</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>8501</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>486</x>
         <y>57</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Port：</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>250</y>
         <width>421</width>
         <height>261</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="title">
        <string>Receive：</string>
       </property>
       <widget class="QTextBrowser" name="textRecv">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>381</width>
          <height>181</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
       </widget>
      </widget>
      <widget class="QPushButton" name="puB_connect">
       <property name="geometry">
        <rect>
         <x>680</x>
         <y>50</y>
         <width>100</width>
         <height>35</height>
        </rect>
       </property>
       <property name="whatsThis">
        <string/>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="AddressEdit">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>50</y>
         <width>151</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>192.168.1.168</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>57</y>
         <width>121</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Address：</string>
       </property>
      </widget>
      <widget class="QPushButton" name="puB_start">
       <property name="geometry">
        <rect>
         <x>680</x>
         <y>143</y>
         <width>100</width>
         <height>35</height>
        </rect>
       </property>
       <property name="text">
        <string>Start</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_plc">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>61</width>
         <height>61</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">border-width: 3px;        
border-radius: 20px;
margin:10px;
padding:20px;
background-color: red;</string>
       </property>
       <property name="inputMethodHints">
        <set>Qt::ImhNone</set>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QPushButton" name="puB_runMode">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>140</y>
         <width>111</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>150</y>
         <width>121</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Run Mode：</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Image viewer</string>
      </attribute>
      <widget class="QPushButton" name="puB_pre">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>430</y>
         <width>101</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>11</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Previous</string>
       </property>
      </widget>
      <widget class="my_qlabel" name="lbl_pic">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>576</width>
         <height>324</height>
        </rect>
       </property>
       <property name="cursor">
        <cursorShape>ArrowCursor</cursorShape>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_pattern">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>380</y>
         <width>251</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Pattern=</string>
       </property>
      </widget>
      <widget class="QPushButton" name="puB_next">
       <property name="geometry">
        <rect>
         <x>450</x>
         <y>430</y>
         <width>111</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Next</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_pic2">
       <property name="geometry">
        <rect>
         <x>600</x>
         <y>70</y>
         <width>259</width>
         <height>194</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>17</pointsize>
        </font>
       </property>
       <property name="cursor">
        <cursorShape>CrossCursor</cursorShape>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_Pos">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>280</y>
         <width>181</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>X = 0 , Y = 0 </string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Command DEMO</string>
      </attribute>
      <widget class="QPushButton" name="puB_read">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>440</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Read</string>
       </property>
      </widget>
      <widget class="QPushButton" name="puB_write">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>530</x>
         <y>440</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Write</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R200">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>80</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R200：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R201">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>170</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R201：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R202">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>260</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R202：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R203">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>350</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R203：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R204">
       <property name="geometry">
        <rect>
         <x>600</x>
         <y>80</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R204：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R205">
       <property name="geometry">
        <rect>
         <x>600</x>
         <y>170</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R205：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R206">
       <property name="geometry">
        <rect>
         <x>600</x>
         <y>260</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R206：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_DM200">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>80</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <stylestrategy>PreferDefault</stylestrategy>
        </font>
       </property>
       <property name="text">
        <string>DM200：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_DM202">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>170</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>DM202：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_DM204">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>260</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>DM204：</string>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_DM206">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>350</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>DM206：</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="DM202_Edit">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>173</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="DM204_Edit">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>263</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="DM206_Edit">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>353</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="DM200_Edit">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>83</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R201_Edit">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>173</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R202_Edit">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>263</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R200_Edit">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>83</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R203_Edit">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>353</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R205_Edit">
       <property name="geometry">
        <rect>
         <x>660</x>
         <y>173</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R206_Edit">
       <property name="geometry">
        <rect>
         <x>660</x>
         <y>263</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLineEdit" name="R204_Edit">
       <property name="geometry">
        <rect>
         <x>660</x>
         <y>83</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="lbl_R207">
       <property name="geometry">
        <rect>
         <x>600</x>
         <y>350</y>
         <width>121</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>R207：</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="R207_Edit">
       <property name="geometry">
        <rect>
         <x>660</x>
         <y>353</y>
         <width>120</width>
         <height>35</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Log</string>
      </attribute>
      <widget class="QComboBox" name="comboBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>161</width>
         <height>25</height>
        </rect>
       </property>
       <property name="currentText">
        <string>select logger</string>
       </property>
       <item>
        <property name="text">
         <string>select logger</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>1</string>
        </property>
       </item>
      </widget>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>10</y>
         <width>701</width>
         <height>481</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="title">
        <string>Log：</string>
       </property>
       <widget class="QTextBrowser" name="text_log">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>671</width>
          <height>421</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>my_qlabel</class>
   <extends>QLabel</extends>
   <header location="global">my_qlabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
